### ISSUE TITLE
Option to utilize ActionDispatch's Directives

### ISSUE THREAD

**Issue by ScotterC** [DocSpring/better_content_security_policy#1]
Thanks for the gem and the write up. It's useful for setting CSP per request. 

I thought I was going to have an issue because your DIRECTIVES don't define `frame_ancestors` which is defined in [ActionDispatch::ContentSecurityPolicy](https://github.com/rails/rails/blob/8015c2c2cf5c8718449677570f372ceb01318a32/actionpack/lib/action_dispatch/http/content_security_policy.rb#L138-L161) , however you took care of that with the method_missing. If you'd like to keep the DIRECTIVES in sync with ActionDispatch you could call `ActionDispatch::ContentSecurityPolicy.const_get(:DIRECTIVES)` and eliminate the method_missing functionality.

https://github.com/DocSpring/better_content_security_policy/blob/174aec5df35c5d0e8e2d9087ff6f8c3b15069ee3/lib/better_content_security_policy/content_security_policy.rb#L8-L25

**Comment by ndbroadbent** [comment]
Thanks for opening the issue! I think this has been addressed in #2, although it would be good to use the Rails DIRECTIVES constant. It's strange that they made it a private constant though:

```
ActionDispatch::ContentSecurityPolicy::DIRECTIVES
NameError: private constant ActionDispatch::ContentSecurityPolicy::DIRECTIVES referenced
```

So I can see why you used `#const_get`, but not sure if it's good to rely on private constants. I wonder if there's a public method or constant we could use

**Comment by ScotterC** [comment]
Agreed. Not in love with the const_get workaround. I didn't see a good reason for it to be private other then it could go away in the future which is a maintenance pain. Still, a CI system based on rails versions would catch it easily.
